opendatakit/briefcase
# Issue 568 improve default export conf panel

Closes #568 

This PR changes the UI layout of the export panel, hiding the default export configuration UI components and replacing them with a button that will open a dialog to set the default conf instead.

![image](https://user-images.githubusercontent.com/205913/43013315-d2246a3a-8c48-11e8-989c-64bb90808a0f.png)

![image](https://user-images.githubusercontent.com/205913/43013346-e9e24f52-8c48-11e8-991d-edb9c20cb6a5.png)

![image](https://user-images.githubusercontent.com/205913/43013359-f31b618a-8c48-11e8-9d3e-407da21a773e.png)

#### What has been done to verify that this works as intended?
From an empty storage dir/clean prefs scenario:
- Pull some forms and go to the export tab
- Verify that the button says "Set Configuration"
- Check the form and verify that the OK button is disabled (pending some conf to be set)
- Click on the "Set Configuration button". The conf dialog opens. Verify that the OK button isn't enabled until I enter the export directory.
- Click on OK. The dialog closes. The button now says "View/Edit Configuration". Verify that the Export button is enabled.
- Relaunch Briefcase
- Verify that the button still says "View/Edit Configuration".
- Click on the button. Verify that the conf is the same I set before relaunching
- Click on "Cancel"
- Select some form and check that the Export button is enabled.
- Click on the "View/Edit Configuration" button.
- Click on "Clear all". The dialog closes. Verify that the button is back to "Set Configuration" and the export button is disabled.
- Relaunch Briefcase
- Verify that the button still says "Set Configuration"
- Click on the "Set Configuration" button and verify it's an empty conf.

#### Why is this the best possible solution? Were any other approaches considered?
We can't continue adding fields to the form because we will eat all the vertical space we need for the forms table.

This is a narrow implementation that reuses much of what we did to support export custom override configurations.

#### Are there any risks to merging this code? If so, what are they?
Nope. Users with saved default configurations should transition without any problem.

#### Does this change require updates to documentation? If so, please file an issue at https://github.com/opendatakit/docs/issues/new and include the link below.
Nope.
